### PR TITLE
More hotkeys

### PR DESCRIPTION
**Pull request by switchlove**
Should address #1454 nicely. Added a fair few additional hotkeys. Added:

```
F - open Farm
  while open
  S - equip/de-equip Shovel
U - open Underground
  while open
  H = select Hammer
  C = select Chisel
  S = try Survey
  B = try Bomb
H - open Hatchery
Shift+S - force save
```

### CODE REVIEW

**Comment by switchlove** [comment]
Failing to build with:
`src\scripts\GameController.ts(206,34): error TS2341: Property 'bomb' is private and only accessible within class 'Mine'.`

Does work in game however, not sure why it doesn't like bomb but survey is fine.

**Review by RedSparr0w — CHANGES_REQUESTED** [review]
Looking good, just a couple of minor changes.

Could you also reset the `docs/index.html` file.
(something along the lines of `git checkout upstream/develop -- docs/index.html`)

> Failing to build with:
> `src\scripts\GameController.ts(206,34): error TS2341: Property 'bomb' is private and only accessible within class 'Mine'.`

Will just need to change the `bomb` function in `Mine.ts` to be `public` and that should fix the error

**Comment by switchlove** [comment]
Lol, totally missed that the E4 was handled. Should be good now.

File: src/scripts/GameController.ts [hunk 1]
```diff
@@ -196,6 +248,30 @@ class GameController {
...
+                                case 4:
+                                    GymRunner.startGym(player.town().gymList[3]);
+                                    break;
+                                case 5:
+                                    GymRunner.startGym(player.town().gymList[4]);
+                                    break;
+                                default:
+                                    return;
+                            }
+                        }
```
[the remark this answers is not in the document]
Comment (switchlove) [reply src/scripts/GameController.ts]: Removed this section

File: src/scripts/GameController.ts [hunk 2]
```diff
@@ -110,8 +120,23 @@ class GameController {
...
+                case 'KeyF':
+                    // Open the Farm with 'F'
+                    if (farms.canAccess() && !$farmsModal.data('disable-toggle')) {
+                        $('.modal').modal('hide');
+                        $farmsModal.data('disable-toggle', true);
+                        $farmsModal.modal('toggle');
+                    }
+                    break;
+                case 'KeyH':
+                    // Open the Hatchery with 'F'
```
Comment (RedSparr0w) [inline src/scripts/GameController.ts:133]: Minor change 😅 
```suggestion
                    // Open the Hatchery with 'H'
```

File: src/scripts/GameController.ts [hunk 3]
```diff
@@ -159,6 +191,26 @@ class GameController {
                             }
                         }
                     }
+                    if ($farmsModal.data('bs.modal')?._isShown) {
+                        if (e.code == 'KeyS') {
+                            FarmController.selectedShovel() ? FarmController.selectedShovel(false) : FarmController.selectedShovel(true);
+                        }
+                    } else if ($undergroundModal.data('bs.modal')?._isShown) {
+                        if (e.code == 'KeyH') {
+                            Mine.toolSelected(Mine.Tool.Hammer);
```
Comment (RedSparr0w) [inline src/scripts/GameController.ts:200]: This doesn't seem to be working for me, it's just bringing up the Hatchery when I try.

**Review by RedSparr0w — CHANGES_REQUESTED** [review]
Just a couple more changes then should be good!

File: src/scripts/GameController.ts (hunk 2 above)
Comment (switchlove) [reply src/scripts/GameController.ts]: Corrected

File: src/scripts/GameController.ts (hunk 3 above)
Comment (switchlove) [reply src/scripts/GameController.ts]: Yeah, did check that after I added the Hatchery. Moved that key into a check in the H case above it.

**Review by RedSparr0w — APPROVED** [review]
Nice work, Thanks for the PR! 🎉

**Pull request merged**